src/package-index/src/target.rs

```diff
@@ -14,36 +15,57 @@ pub fn package_target() -> Result<Target, Error> {
     Ok(target)
 }
 
-#[derive(Debug, Copy, Clone, Hash, PartialEq, Eq, Serialize, Deserialize)]
-#[allow(non_camel_case_types)]
-pub enum Target {
-    #[serde(rename = "x86_64-apple-darwin")]
-    X86_64AppleDarwin,
-    #[serde(rename = "x86_64-unknown-linux-musl")]
-    X86_64UnknownLinuxMusl,
-}
+/// An object representing a specific build target for an artifact
+/// being managed by fluvio-index.
+///
+/// This type is generally constructed using `FromStr` via the
+/// `parse` method.
+///
+/// # Example
+///
+/// ```
+/// # use fluvio_index::Target;
+/// let target: Target = "x86_64-unknown-linux-musl".parse().unwrap();
+/// ```
+#[derive(Debug, Clone, Hash, PartialEq, Eq, Serialize)]
+#[serde(transparent)]
+pub struct Target(Cow<'static, str>);
```

Comment:
Hmm, I kinda like the enum of this better. I understand why you've done it this way though. Can one use the `transparent` and `rename` with serde at the sametime?

Reply:
What is it about the enum that you like better? With the enum representation we would need to add variants for each new target we want to support. In either case, the "serialized" representation is the same, it is transparently represented as a string that is just the name of the target.

What do you mean about `rename`? I'm no longer using `rename` in this new version.

Reply:
@simlay ?

Reply:
Given that we are going to add more variants, I don't think we can encode as Enum

Reply:
Sounds good. Let's keep as is.